src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -929,6 +961,25 @@ private void setupCustomDeviceMapping(List<BlockDeviceMapping> deviceMappings) {
                 }
                 String slaveName = spotInstReq.getSpotInstanceRequestId();
 
+                if (spotConfig.fallbackToOndemand) {
+                    for (int i = 0; i < 2 && spotInstReq.getStatus().getCode().equals("pending-evaluation"); i++) {
```

Comment:
I guess it's not exactly required, but jenkins-core somewhat recently added a `Retrier<T>` class which might be useful here. Might be worth looking at? I'm mostly concerned about `Thread.sleep` calls for testing purposes.

Reply:
hmm, the Retrier is very nice. However, it requires us to bump the jenkins version to at least 2.152. And the Retrier is marked `NoExternalUse` which crashes the build (unless there's a way around that).

Reply:
Oh right. NoExternalUse is for avoiding creating new public APIs that can never change again. Ah well.